src/main/java/frc/robot/utils/VisionDataProcessor.java

```diff
@@ -0,0 +1,80 @@
+package frc.robot.utils;
+
+import frc.robot.RobotConstants;
+import frc.robot.pose.TargetLocation;
+import frc.robot.pose.VisionData;
+import java.util.*;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+
+// Template for data through serial port is
+// boolean(targetfound) int(x) int(y) int(targetwidth) double(framerate) -\n
+
+public class VisionDataProcessor {
+
+    private String buffer = "";
+    private String retval;
+    private int visionDataMemory = 5;
+    
+    private FixedStack visionDataStack = new FixedStack(visionDataMemory);
+    
+    public VisionDataProcessor() {
+        visionDataStack.add(RobotConstants.ROBOT_DEFAULTS.VISION.DEFAULT_VISION_DATA);
+    }
+
+    public TargetLocation compute(VisionData vData) {
+        double distance = 0;
+        double angle = 0;
+        //Code here from empirical testing
+        return new TargetLocation(distance, angle);
+    }
+
+    public void addInput(String readings){
+        buffer += readings;
+    }
+    
+    private boolean completeData(String data){
+        if(data.length() == 0){
+            return false;
+        }
+        return data.substring(data.length() - 1).equals("-");
+    }
+    
+    private void parseBuffer(){
+        String[] dataEntriesBuffer = buffer.split("\n");
+        for(String data: dataEntriesBuffer){
+            if(!completeData(data)){
+                continue;
+            }
+            visionDataStack.add(calculateVisionData(data));
+        }
+        if(buffer.lastIndexOf("\n") != -1){
+            buffer = buffer.substring(buffer.lastIndexOf("\n"));
+        }
+    }
+    
+    public VisionData getCurrentVisionData(){
+        parseBuffer();
+        return visionDataStack.peek();
+    }
+     
+    private VisionData calculateVisionData(String readings) {
+        String[] outputData = readings.split(" ");
+        boolean visionTargetFound = Boolean.parseBoolean(outputData[0]);
+
+        double lateralOffset = -1;
```

Comment:
a standard and better way to do this is to define (probably in VisionData), a static field that contains a VisionData that is set to the special 'no target' values, like:

public static VisionData NO_TARGET = new VisionData(-1, -1, -1);

Then, below you do something like:

if ( visiontTargetFound){
   return new VisionData(...)
else{
  return VisionData.NO_TARGET
}

that gets rid of the magic numbers here, while also making it possible for others to know what value they should use for 'no data'